internal/interactors/create-job.go

```diff
@@ -0,0 +1,117 @@
+package interactors
+
+import (
+	"fmt"
+	"job-worker/internal/dto"
+	jobEntity "job-worker/internal/models/job"
+	"job-worker/internal/repository"
+	"job-worker/pkg/worker"
+	"log"
+	"time"
+)
+
+func CreateJob(createJobRequest dto.CreateJobRequest) (*dto.CreateJobResponse, error) {
+	job := createJobRequest.ToJob()
+
+	process, err := createWorkerProcess(createJobRequest.Command, job.ID)
+	if err != nil {
+		log.Printf("could not create process %s\n", err)
+		return nil, err
+	}
+
+	savedJob, err := persistJob(job, process)
+	if err != nil {
+		log.Printf("could not persist job %s\n", err)
+		return nil, err
+	}
+
+	err = process.Start()
+	if err != nil {
+		log.Printf("could not start process %s\n", err)
+		setJobStatusFailed(savedJob)
+		return nil, fmt.Errorf("couldn't start process for job %s with error: %s", savedJob.ID, err)
+	}
+
+	setJobStatusRunning(savedJob)
+	go waitForExitReason(savedJob, process)
+
+	return &dto.CreateJobResponse{ID: savedJob.ID}, nil
+}
+
+func waitForExitReason(job *jobEntity.Job, process *worker.Process) {
+	exitReason := <-process.ExitChannel
+
+	switch exitReason.ExitCode {
+	case -1:
+		finishJobWithStatusAndCode(job, jobEntity.STOPPED, exitReason.ExitCode)
+	case 0:
+		finishJobWithStatusAndCode(job, jobEntity.COMPLETED, exitReason.ExitCode)
+	case 1:
+		finishJobWithStatusAndCode(job, jobEntity.FAILED, exitReason.ExitCode)
```

Comment:
nit: remove case since it's covered by default, or...
```suggestion
	case 1:
		fallthrough
```

Reply:
Done here https://github.com/lucasbfernandes/job-worker/commit/067e9c5dc06e8f28d1668d244af039ac7619ca01